R/DBConnection.R

```diff
@@ -106,6 +106,41 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' @export
+dbBreak <- structure(list(), class = "dbiAbort")
```

Comment:
Ah, I think I understand better your objection now. You're right that the particular way I had designed the function wasn't working quite right. (I hadn't realized that `tryCatch` would only be triggered in response to an existing condition -- i.e. `stop()`, `warning()` or `message()`.) So I updated it (and actually simplified it!) to work how it's supposed to, which is analogous to a regular `break` statement. The implementation details are equally hidden, I think. Instead of telling the user to call a specific function to break out the the function, we tell them to return a specific value.

So, again, I think we're at the same impasse as before -- do we want this to be analogous to a control statement or, rather, a regular function that raises a condition?

As for the custom error message, I think it's the same thing: if `dbBreak` functions as `break` does, it doesn't make sense for it to be associated with an error message. In fact, in that case, it shouldn't be seen as an error at all -- rather, it's the user expectedly controlling the execution of the code using a control statement, rather than unexpectedly interrupting the execution of the code because of an error.

Since this conception of how `dbBreak` was to work was the result of a discussion with @jcheng5 (and I think @hadley, though I can't quite recall), I'd prefer to get their input before we change this. However, if you feel strongly about this and no-one else pitches in, I'm okay changing things over. I personally think this is a more intuitive and idiomatic way to code a function like this, but I'm not strongly opposed to the alternative.